posthog/batch_exports/service.py

```diff
@@ -335,18 +340,26 @@ def backfill_export(
         batch_export_id: The id of the BatchExport to backfill.
         team_id: The id of the Team the BatchExport belongs to.
         start_at: From when to backfill.
-        end_at: Up to when to backfill.
+        end_at: Up to when to backfill, if None it will backfill until it has caught up with realtime
+                and then unpause the underlying BatchExport.
     """
     try:
-        BatchExport.objects.get(id=batch_export_id, team_id=team_id)
+        batch_export = BatchExport.objects.select_related("destination").get(id=batch_export_id, team_id=team_id)
     except BatchExport.DoesNotExist:
         raise BatchExportIdError(batch_export_id)
 
+    # Ensure we don't allow users access to this feature until we are ready.
+    if not end_at and batch_export.destination.type not in (
+        BatchExportDestination.Destination.HTTP,
+        BatchExportDestination.Destination.NOOP,  # For tests.
+    ):
+        raise BatchExportWithNoEndNotAllowedError(f"BatchExport {batch_export_id} has no end_at and is not HTTP")
```

Comment:
Not sure if we are doing this already, but we may be want to return a 400 from this?